### ISSUE TITLE
Revisit MiAIRR non-nullable fields

### ISSUE THREAD

**Issue by bcorrie** [airr-community/airr-standards#310]
Creating this issue to replace the specific issue around cell_storage in #307 (which I will close).

In the current spec the following are required and non-nullable. Note ```sample_type```, ```tissue```, and ```single_cell```  are missing from the original list in #96

Are these still the "correct" fields to be required and non-nullable?

```
        synthetic:
            type: boolean
            description: TRUE for libraries in which the diversity has been synthetically generated (e.g. phage display)
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        organism:
            $ref: '#/Ontology'
            description: Binomial designation of subject's species
            example:
                id: 9096
                value: Homo sapiens
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        cell_storage:
            type: boolean
            description: TRUE if cells were cryo-preserved between isolation and further processing
            example: TRUE
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        template_class:
            type: string
            enum:
                - DNA
                - RNA
            description: The class of nucleic acid that was used as primary starting material for the following procedures
            example: RNA
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        library_generation_method:
            type: string
            enum:
                - "PCR"
                - "RT(RHP)+PCR"
                - "RT(oligo-dT)+PCR"
                - "RT(oligo-dT)+TS+PCR"
                - "RT(oligo-dT)+TS(UMI)+PCR"
                - "RT(specific)+PCR"
                - "RT(specific)+TS+PCR"
                - "RT(specific)+TS(UMI)+PCR"
                - "RT(specific+UMI)+PCR"
                - "RT(specific+UMI)+TS+PCR"
                - "RT(specific)+TS"
                - "other"
            description: Generic type of library generation
            example: RT(oligo-dT)+TS(UMI)+PCR
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        complete_sequences:
            type: string
            enum:
                - partial
                - complete
                - "complete+untemplated"
            description: >
                To be considered `complete`, the procedure used for library construction MUST generate sequences that
                1) include the first V segment codon that encodes the mature polypeptide chain (i.e. after the
                leader sequence) and 2) include the last complete codon of the J segment (i.e. 1 bp 5' of the J->C
                splice site) and 3) provide sequence information for all positions between 1) and 2). To be considered
                `complete & untemplated`, the sections of the sequences defined in points 1) to 3) of the previous
                sentence MUST be untemplated, i.e. MUST NOT overlap with the primers used in library preparation.
            example: partial
            x-airr:
                miairr: true
                required: true
                nullable: false
```
```
        physical_linkage:
            type: string
            enum:
                - none
                - "hetero_head-head"
            description: >
                Describes the mode of linkage if a method was used which physically links nucleic acids derived from
                distinct loci in a single-cell context.
            example: hetero_head-head
            x-airr:
                miairr: true
                required: true
                nullable: false
```

**Comment by bcorrie** [comment]
@bussec comment from #307: 

> I understand that this makes it hard to third-party annotators, if this information is not passed on in the original publication. We can discuss whether nullable:false should be reserved for those cases where that have an undeniable relevance for the interpretation of the study (i.e. species). The original list of non-nullable fields is in #96 (comment) . We will discuss this in the next MiniStd call, all input welcome.

**Comment by javh** [comment]
From the call:
* Could be pushed back to v2.0.

**Comment by bussec** [comment]
Going through the fields again, there are two changes that I would consider reasonable:

1. Make `cell_storage` nullable (it is important to know, but IMO not in the same league as the others)
2. Add an further `mixed` option to `complete_sequences`, but keep the non-nullable status

Everything else really seems to be critical for interpretation and complete in term of enumerations. The one exception that we should discuss is whether there is any scenario in which the boolean option for `synthetic` would not work.

**Comment by bussec** [comment]
@bcorrie when we fix this, we should fix #204 en passant...

**Comment by bcorrie** [comment]
> @bcorrie when we fix this, we should fix #204 en passant...

I concur, any other issues we can raise as ......... other issues...

**Comment by bcorrie** [comment]
@bussec any reason we don't create a pull request for two minor changes you suggest above and merge these before v1.3?

**Comment by bussec** [comment]
@bcorrie see above PR, which is complete by now. I don't have strong opinions whether this should go into v1.3 or v2.0. Technically there was a change to a MiAIRR field name (for the `physical linkage` key) so it would be better for v2.0, but as there is -- best to my knowledge -- no submission using this field yet, we could push it to 1.3, too.

**Comment by bcorrie** [comment]
I would say that if we are done with them I would prefer to have them in v1.3. There isn't a change to the physical_linkage field name, just the controlled vocabulary, correct. And there is an addition to the possible terms rather than a deletion, so I think it is OK for v1.3

**Comment by bussec** [comment]
Closed via #319.
